ably/realtime_channel.go

```diff
@@ -279,6 +305,23 @@ func (c *RealtimeChannel) detachUnsafe() (result, error) {
 	return res, nil
 }
 
+func (c *RealtimeChannel) detachUnsafe() (result, error) {
+	if c.state == ChannelStateAttaching { //RTL5i
```

Comment:
Why isn't this part of the `detach` function? All other checks are there and `detachUnsafe` is always called from there, except possibly on reconnection, in which case the channel will be always DETACHING.

Reply:
Okay, if that's the case where for reconnection channel state is always detaching, I will move it to detach function